src/libmongoc/tests/TestSuite.c

```diff
@@ -276,61 +278,53 @@ TestSuite_AddLive(TestSuite *suite, /* IN */
                   const char *name, /* IN */
                   TestFunc func)    /* IN */
 {
-   TestSuite_AddFullWithTestFn(suite, name, TestSuite_AddHelper, NULL, func, TestSuite_CheckLive);
-}
-
-
-static void
-_TestSuite_AddCheck(Test *test, CheckFunc check, const char *name)
-{
-   test->checks[test->num_checks] = check;
-   if (++test->num_checks > MAX_TEST_CHECK_FUNCS) {
-      MONGOC_STDERR_PRINTF("Too many check funcs for %s, increase MAX_TEST_CHECK_FUNCS "
-                           "to more than %d\n",
-                           name,
-                           MAX_TEST_CHECK_FUNCS);
-      abort();
-   }
+   // Add the `lock:live-server` tag to the test.
+   mstr with_tags = mstr_sprintf("%s [lock:live-server]", name);
+   TestSuite_AddFullWithTestFn(suite, with_tags.data, TestSuite_AddHelper, NULL, func, TestSuite_CheckLive);
+   mstr_destroy(&with_tags);
 }
 
 
 Test *
-_V_TestSuite_AddFull(TestSuite *suite, const char *name, TestFuncWC func, TestFuncDtor dtor, void *ctx, va_list ap)
+_V_TestSuite_AddFull(
+   TestSuite *suite, const char *name_and_tags, TestFuncWC func, TestFuncDtor dtor, void *ctx, va_list ap)
 {
-   CheckFunc check;
-   Test *test;
-   Test *iter;
-
-   if (suite->ctest_run && (0 != strcmp(suite->ctest_run, name))) {
+   // Split the name and tags around the first whitespace:
+   mstr_view name, tags;
+   mstr_split_around(mstr_cstring(name_and_tags), mstr_cstring(" "), &name, &tags);
+   // Trim extras:
+   tags = mstr_trim(tags);
+
+   if (suite->ctest_run.data && mstr_cmp(suite->ctest_run, !=, name)) {
+      // We are running CTest, and not running this particular test, so just skip registering it
       if (dtor) {
          dtor(ctx);
       }
       return NULL;
    }
 
-   test = (Test *)bson_malloc0(sizeof *test);
-   test->name = bson_strdup(name);
+   Test *test = TestVec_push(&suite->tests);
+   test->name = mstr_copy(name);
    test->func = func;
-   test->num_checks = 0;
 
+   mstr_view tag, tail;
+   tail = tags;
+   while (tail.len) {
+      mlib_check(mstr_split_around(tail, mstr_cstring("["), NULL, &tag),
+                 because,
+                 "Expected an opening bracket for the next test tag");
+      mlib_check(mstr_split_around(tag, mstr_cstring("]"), &tag, &tail), because, "Expected a closing bracket for tag");
```

Comment:
Minor issue: this message is emitted even for cases such as `"/bson/error/basic invalid]"`:

```
TestSuite.c:315: in [_V_TestSuite_AddFull]: Check condition ⟨mstr_split_around(((mstr_view) {(((tail)).data), (((tail)).len)}), ((mstr_view) {(((mstr_cstring("["))).data), (((mstr_cstring("["))).len)}), (((void*)0)), (&tag))⟩ failed (Expected an opening bracket for the next test tag)
```

Perhaps reword the error message to be less specific.

Reply:
Clarified wording in assertion message to explain the cause being the whitespace introducing a tag list.